.gitignore

```diff
@@ -44,5 +44,32 @@ reporting/*.pdf
 reporting/*.txt
 reporting/cookies.json
 
-#IDEs
-.idea/
+# Idea
+.idea/**/workspace.xml
+.idea/**/tasks.xml
+.idea/**/usage.statistics.xml
+.idea/**/dictionaries
+.idea/**/shelf
+.idea/**/contentModel.xml
+.idea/**/dataSources/
+.idea/**/dataSources.ids
+.idea/**/dataSources.local.xml
+.idea/**/sqlDataSources.xml
+.idea/**/dynamic.xml
+.idea/**/uiDesigner.xml
+.idea/**/dbnavigator.xml
+.idea/**/gradle.xml
+.idea/**/libraries
+.idea/**/mongoSettings.xml
+.idea/httpRequests
+.idea/caches/build_file_checksums.ser
+.idea/replstate.xml
+.idea_modules/
+cmake-build-*/
+*.iws
+out/
+atlassian-ide-plugin.xml
+com_crashlytics_export_strings.xml
+crashlytics.properties
+crashlytics-build.properties
+fabric.properties
```

Comment:
Modern IntelliJ products include its own .gitignore in the `.idea` folder where they already added all the necessary files to ignore.

```suggestion
```

Reply:
But do we really need to ignore `cmake-build`, `gradle.xml` and so on? 
I mean if IDE for some reason creates them we better know why. If we just ignore them we won't know.
And if we include Java-related ignores why not include Python, PHP, Ruby etc? 

Default `.gitignore` already in the `.idea` folder on my machine produces almost the same file list to commit as in this PR... so it's doing OK.

Reply:
> And if we include Java-related ignores why not include Python, PHP, Ruby etc?

Idea is created using Java, were it being created using Python, there would be Python references in what the IDE files save in `.idea/`.


> But do we really need to ignore cmake-build, gradle.xml and so on?

The original ignore was a single line 
```
.idea
```
and going from it to include almost anything save for
 ```
/shelf/
/workspace.xml
# Editor-based HTTP Client requests
/httpRequests/
```
is an open invitation for unforeseen consequences (HL2 pun).

So, to the question of
> do we really need to ignore `gradle.xml` at al?

 I can offer only the phrase 
> better safe than sorry

which is perfectly accomplished by the original ignore file.

Do note that `gradle.xml` can pop in at any subdirectory (or several of) under `.idea/` so one can make the assumption IDE or plugins for it can create it during the course of their work.

>  we better know why. If we just ignore them we won't know.

I'd prefer not to know why the hard way by making a mess, then cleaning it up, then adding the files to ignore.
But, as this is a draft PR and there is a branch associated to it, I can welcome you to test and provide a reason why they shouldn't be ignored.

For my part, I can only go so far as create a Gist of the common groups using the settings of "allow duplicate lines" and "allow comments and empty lines":

https://gist.github.com/goran-peoski-work/5249e1982b59573149d01b3fb73952fd

![sparkle-ignore-03](https://user-images.githubusercontent.com/79229621/128981044-78c4ea64-0015-41c9-8b9a-f226840c23f0.png)


On the note of the empirical data and the conclusion from it: 
> Default .gitignore already in the .idea folder on my machine produces almost the same file list to commit as in this PR... so it's doing OK.

Extrapolation of one example, some times colloquially phrased as 
> it works on my machine

better illustrated with the act of scooping a glass of water from the ocean, observing it and concluding there are no sharks in the water 🤷